sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +158,306 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
- * @brief                  Appends one uint8_t value to the #az_span if there is enough capacity for
- * it.
- * @param[in]  span        #az_span where to append the char.
- * @param[in]  c           Value to be appended.
- * @param[out] out_span    #az_span returned as result of the concatenation.
- * @return                 An #az_result value indicating the result of the operation.
- *                         #AZ_OK                      If no errors occurs.
- *                         #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY   If there is not enough capacity left to append the `c`.
+ * @brief The AZ_PAIR_NULL macro returns an az_pair instance whose key and value fields are
+ * initialized to AZ_SPAN_NULL
+ *
  */
-AZ_NODISCARD az_result az_span_append_uint8(az_span span, uint8_t c, az_span* out_span);
+#define AZ_PAIR_NULL \
+  (az_pair) { .key = AZ_SPAN_NULL, .value = AZ_SPAN_NULL }
 
 /**
- * @brief                 Appends one uint32_t to the #az_span if there is enough capacity for it.
- * @param[in]  span       #az_span where to append the value.
- * @param[in]  n          Value to be appended.
- * @param[out] out_span   Pointer where to store the resulting #az_span.
- * @return                An #az_result value indicating the result of the operation.
- *                        #AZ_OK                      If no errors occurs.
- *                        #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY   If there is not enough capacity left to append the `n`.
+ * @brief az_pair_init returns an az_pair with its key and value fields initialized to the specified
+ * key and value parameters.
+ *
+ * @param[in] key A span whose bytes represent the key
+ * @param[in] value A span whose bytes represent the key's value
+ * @return  An az_pair with the field initialized to the parameters' values
  */
-AZ_NODISCARD az_result az_span_append_u32toa(az_span span, uint32_t n, az_span* out_span);
+AZ_NODISCARD AZ_INLINE az_pair az_pair_init(az_span key, az_span value)
+{
+  return (az_pair){ .key = key, .value = value };
+}
 
 /**
- * @brief                 Appends one int32_t to the #az_span if there is enough capacity for it.
- * @param[in]  span       #az_span where to append the value.
- * @param[in]  n          Value to be appended.
- * @param[out] out_span   Pointer where to store the resulting #az_span.
- * @return                An #az_result value indicating the result of the operation.
- *                        #AZ_OK                      If no errors occurs.
- *                        #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY   If there is not enough capacity left to append the `n`.
+ * @brief az_pair_from_str returns an az_pair with its key and value fields initialized to span's
+ * over the specified key and value 0-terminated string parameters.
+ *
+ * @param[in] key A string representing the key
+ * @param[in] value A string representing the key's value
+ * @return  An az_pair with the field initialized to the az_span instances over the passed-in strings
```

Comment:
Should we mention what the length/capacity of each span will be within the pair?

Reply:
I don't think this is necessary as this is the only thing that would make any sense.